gomock/matchers.go

```diff
@@ -382,6 +402,18 @@ func Not(x any) Matcher {
 	return notMatcher{Eq(x)}
 }
 
+// Regex checks whether parameter matches the associated regex.
+//
+// Example usage:
+//
+// Regex("[0-9]{2}:[0-9]{2}").Matches("23:02") // returns true
+// Regex("[0-9]{2}:[0-9]{2}").Matches([]byte{'2', '3', ':', '0', '2'}) // returns true
+// Regex("[0-9]{2}:[0-9]{2}").Matches("hello world") // returns false
+// Regex("[0-9]{2}").Matches(21) // returns false as it's not a valid type
```

Comment:
```suggestion
//	Regex("[0-9]{2}:[0-9]{2}").Matches("23:02") // returns true
//	Regex("[0-9]{2}:[0-9]{2}").Matches([]byte{'2', '3', ':', '0', '2'}) // returns true
//	Regex("[0-9]{2}:[0-9]{2}").Matches("hello world") // returns false
//	Regex("[0-9]{2}").Matches(21) // returns false as it's not a valid type
```
nit: let's indent these so they render in the docs correctly.